Editor/Window/GameLiftRequestAdapter.cs

```diff
@@ -0,0 +1,242 @@
+using UnityEngine;
+using UnityEngine.UIElements;
+
+namespace Editor.Window
+{
+    public class GameLiftRequestAdapter
+    {
+        private GameLiftPlugin _gameLiftConfig;
+        private string _fleetName;
+        private string _fleetId;
+        public const string FleetLocation = "custom-location-1";
+        private const string FleetDescription = "Deployed by the Amazon GameLift Plug-in for Unity.";
+        private VisualElement _container;
+
+        public GameLiftRequestAdapter(GameLiftPlugin gameLiftConfig)
+        {
+            _gameLiftConfig = gameLiftConfig;
+        }
+
+        internal async Task<bool> CreateAnywhereFleet(string fleetName)
+        {
+            if (_gameLiftConfig.GameLiftWrapper != null)
+            {
+                var success = await CreateCustomLocationIfNotExists(FleetLocation);
+                if (success)
+                {
+                    var fleetId = await CreateFleet(ComputeType.ANYWHERE, FleetLocation,fleetName);
+                    var fleetNameResponse = _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.FleetName, fleetName);
+                    var fleetIdResponse = _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.FleetId, fleetId);
+                    var customLocationNameResponse =
+                        _gameLiftConfig.CoreApi.PutSetting(SettingsKeys.CustomLocationName, FleetLocation);
+                    return fleetNameResponse.Success && customLocationNameResponse.Success && fleetIdResponse.Success;
+                }
+
+                return false;
+            }
+
+            Debug.Log("Error: Missing Account Profile");
+            return false;
+        }
+        
+        private async Task<bool> CreateCustomLocationIfNotExists(string fleetLocation)
+        {
+            try
+            {
+                var listLocationsResponse = await _gameLiftConfig.GameLiftWrapper.ListLocations(new ListLocationsRequest
+                {
+                    Filters = new List<string> { "CUSTOM" }
+                });
+
+                var foundLocation =
+                    listLocationsResponse.Locations.FirstOrDefault(l => l.LocationName.ToString() == fleetLocation);
+
+                if (foundLocation == null)
+                {
+                    var createLocationResponse = await _gameLiftConfig.GameLiftWrapper.CreateLocation(
+                        new CreateLocationRequest()
+                        {
+                            LocationName = fleetLocation
+                        });
+
+                    if (createLocationResponse.HttpStatusCode == HttpStatusCode.OK)
+                    {
+                        Debug.Log($"Created Custom Location {fleetLocation}");
+                    }
+                }
+
+                return true;
+            }
+            catch (Exception ex)
+            {
+                var errorBox = _container.Q<VisualElement>("FleetErrorInfoBox");
+                errorBox.style.display = DisplayStyle.Flex;
```

Comment:
Could you help understand what this is doing? Is this something that can be done in CSS instead?

Reply:
This displays the error box that is in the mocks to the user with the appropriate message.

Reply:
Could this be done in CSS instead? This is the same code we had refactored out in GameLiftRequestAdapter.cs, can we just use the same utility instead?

Reply:
It could be moved to being controlled be a css class, toggling it on and off instead of changing inline styles